### PR TITLE
LPS-133337 Update InstanceWorkflowMetricsIndexer's integration tests

### PR DESCRIPTION
**Pull request by MarinhoFeliphe**

### CODE REVIEW

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/resources/com/liferay/portal/workflow/metrics/internal/search/index/dependencies/workflow-metrics-update-task-script.painless [hunk 1]
```diff
@@ -4,5 +4,6 @@ if (ctx._source.containsKey("tasks")) {
 	if (task != null) {
 		task.assigneeIds = params.assigneeIds;
 		task.assigneeType = params.assigneeType;
+		task.assigneeName = params.assigneeName;
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/resources/com/liferay/portal/workflow/metrics/internal/search/index/dependencies/workflow-metrics-update-task-script.painless:7]: Hey @MarinhoFeliphe, sort this. please

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/search/index/TaskWorkflowMetricsIndexerImpl.java [hunk 2]
```diff
@@ -307,26 +309,38 @@ public Document updateTask(
...
+							"script.painless")
+				).language(
+					"painless"
+				).putParameter(
+					"assigneeIds", assigneeIds
+				).putParameter(
+					"assigneeType", assigneeType
+				);
+
+				if (Objects.equals(assigneeType, User.class.getName()) &&
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/search/index/TaskWorkflowMetricsIndexerImpl.java:335]: Hey @MarinhoFeliphe, put the if statement after `"assigneeIds", assigneeIds`(alphabetical order). please.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/search/index/TaskWorkflowMetricsIndexerImpl.java [hunk 3]
```diff
@@ -134,6 +134,25 @@ public Document addTask(
...
+					HashMapBuilder.<String, Object>put(
+						"assigneeIds", assigneeIds
+					).put(
+						"assigneeType", assigneeType
+					).put(
+						"taskId", taskId
+					).put(
+						"taskName", name
+					);
+
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/search/index/TaskWorkflowMetricsIndexerImpl.java:147]: Hey @MarinhoFeliphe as you are using a HashMapBuilder here. It is possible to keep the `HashMapBuilder` in an inline declaration. To achieve this, you need to use the put option that accepts a `BaseMapBuilder.UnsafeSupplier<V, Exception> valueUnsafeSupplier` and return null when it is not a `User`.
Thx

Comment (rafaprax) [reply modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/search/index/TaskWorkflowMetricsIndexerImpl.java:147]: And keep the order here as well 👍

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/model/listener/UserModelListener.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+		policy = ReferencePolicy.DYNAMIC,
+		policyOption = ReferencePolicyOption.GREEDY,
+		target = "(search.engine.impl=Elasticsearch)"
+	)
+	protected volatile SearchEngineAdapter searchEngineAdapter;
+
+	@Reference(target = "(workflow.metrics.index.entity.name=instance)")
+	private WorkflowMetricsIndex _instanceWorkflowMetricsIndex;
+
+	private boolean _nameModified;
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/model/listener/UserModelListener.java:116]: Hey @MarinhoFeliphe do not use an stateful approach here. It'll be error prone when this ModelListener receives concurrent accesses(more than one thread at once). Please use a stateless approach(just move the code to the same method).


```java
@Override
public void onBeforeUpdate(User user) throws ModelListenerException {
  User currentUser = _userLocalService.fetchUserById(user.getUserId());
  
  TransactionCommitCallbackUtil.registerCallback(
    () -> {
    
      if(!Objects.equals(currentUser.getFullName(), user.getFullName())) {
          //execute update here
       }
    });
  }
```

Thx

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/model/listener/UserModelListener.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+/**
+ * @author Feliphe Marinho
+ */
+@Component(immediate = true, service = ModelListener.class)
+public class UserModelListener extends BaseModelListener<User> {
+
+	@Override
+	public void onAfterUpdate(User user) throws ModelListenerException {
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/model/listener/UserModelListener.java:50]: @MarinhoFeliphe you don't need to user the `onAfterUpdate` cycle. Move it to `onBeforeUpdate`. Thx

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/model/listener/UserModelListener.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+					Object[]::new
+				));
+
+			nestedBooleanQuery.addMustQueryClauses(termsQuery);
+
+			nestedBooleanQuery.addMustQueryClauses(
+				_queries.term("tasks.assigneeType", User.class.getName()));
+
+			_workflowMetricsPortalExecutor.execute(
+				() -> searchEngineAdapter.execute(
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/model/listener/UserModelListener.java:88]: @MarinhoFeliphe move the whole logic to create the UpdateByQueryDocumentRequest to the `Callable`.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/resources/com/liferay/portal/workflow/metrics/internal/model/listener/dependencies/workflow-metrics-update-task-assignee-script.painless [hunk 7]
```diff
@@ -0,0 +1,7 @@
+if (ctx._source.containsKey("tasks")) {
+	for (task in ctx._source.tasks) {
+		if (task != null && task.assigneeIds.contains(params.userId)) {
+    		task.assigneeName = params.userName;
+    	}
+	}
+}
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/resources/com/liferay/portal/workflow/metrics/internal/model/listener/dependencies/workflow-metrics-update-task-assignee-script.painless:7]: @MarinhoFeliphe format this code properly. Thx

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/internal/search/index/test/InstanceWorkflowMetricsIndexerTest.java [hunk 8]
```diff
@@ -115,8 +128,86 @@ public void testReindex() throws Exception {
 			workflowDefinition.getWorkflowDefinitionId());
 	}
 
+	@Test
+	public void testUpdateAssigneeUser() throws Exception {
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/internal/search/index/test/InstanceWorkflowMetricsIndexerTest.java:132]: @MarinhoFeliphe What do you think to break this in two tests? And do not user `AssigneeUser` in the method name. I think that just `blaBlaBlaAssignee` is enough.

Thx

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/internal/search/index/test/InstanceWorkflowMetricsIndexerTest.java [hunk 9]
```diff
@@ -115,8 +128,86 @@ public void testReindex() throws Exception {
...
+		User user = UserTestUtil.addUser(
+			RandomTestUtil.randomString(
+				NumericStringRandomizerBumper.INSTANCE,
+				UniqueStringRandomizerBumper.INSTANCE),
+			LocaleUtil.getDefault(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(),
+			new long[] {TestPropsValues.getGroupId()});
+
+		List<KaleoTaskInstanceToken> kaleoTaskInstanceTokens =
+			_kaleoTaskInstanceTokenLocalService.getKaleoTaskInstanceTokens(
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/internal/search/index/test/InstanceWorkflowMetricsIndexerTest.java:159]: Hey @MarinhoFeliphe Can we move this to the `assignKaleoTaskInstanceToken` method somehow?
Thx

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/internal/search/index/test/InstanceWorkflowMetricsIndexerTest.java [hunk 10]
```diff
@@ -115,8 +128,86 @@ public void testReindex() throws Exception {
...
+			kaleoInstance.getClassName(), "classPK", kaleoInstance.getClassPK(),
+			"companyId", kaleoInstance.getCompanyId(), "completed", false,
+			"deleted", false, "instanceId", kaleoInstance.getKaleoInstanceId(),
+			"processId", workflowDefinition.getWorkflowDefinitionId(),
+			"version", "1.0");
+
+		user.setMiddleName(RandomTestUtil.randomString());
+
+		_userLocalService.updateUser(user);
+
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/internal/search/index/test/InstanceWorkflowMetricsIndexerTest.java:185]: Hey @MarinhoFeliphe please reassign the `user` variable or create a new one to receive the result of the `updateUser` call;

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/util/BaseWorkflowMetricsIndexerTestCase.java [hunk 11]
```diff
@@ -226,6 +226,26 @@ protected void assertSLAReindex(
 		assertSLAReindex(indexNamesMap, indexTypes, parameters);
 	}
 
+	protected KaleoTaskInstanceToken assignKaleoTaskInstanceToken(
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/util/BaseWorkflowMetricsIndexerTestCase.java:229]: Hello @rafaprax,

I put `getKaleoTaskInstanceTokens` method this way, what do you think?

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/resources/com/liferay/portal/workflow/metrics/internal/model/listener/dependencies/workflow-metrics-update-task-assignee-script.painless [hunk 12]
```diff
@@ -0,0 +1,7 @@
+if (ctx._source.containsKey("tasks")) {
+	for (task in ctx._source.tasks) {
+		if (task != null && task.assigneeIds.contains(params.userId)) {
+    		task.assigneeName = params.userName;
+		}
+	}
+}
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/resources/com/liferay/portal/workflow/metrics/internal/model/listener/dependencies/workflow-metrics-update-task-assignee-script.painless:7]: Hey @MarinhoFeliphe it still not formatted correctly.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/util/BaseWorkflowMetricsIndexerTestCase.java (hunk 11 above)
Comment (rafaprax) [reply modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/util/BaseWorkflowMetricsIndexerTestCase.java:229]: It's ok to me

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/util/BaseWorkflowMetricsTestCase.java [hunk 13]
```diff
@@ -77,6 +80,54 @@ public void tearDown() throws Exception {
...
+				"Parameters length is not an even number");
+		}
+
+		CountSearchRequest countSearchRequest = new CountSearchRequest();
+
+		countSearchRequest.setIndexNames(indexName);
+
+		BooleanQuery booleanQuery = queries.booleanQuery();
+
+		BooleanQuery filterQuery = queries.booleanQuery();
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/util/BaseWorkflowMetricsTestCase.java:107]: @MarinhoFeliphe rename to filterBooleanQuery, please.

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/model/listener/UserModelListener.java [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+
+							termsQuery.addValues(
+								Stream.of(
+									user.getUserId()
+								).map(
+									String::valueOf
+								).toArray(
+									Object[]::new
+								));
+
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/model/listener/UserModelListener.java:92]: Instantiate a `Object` array directly here.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:forward

**Review by rafaprax — APPROVED** [review]

**Comment by rafaprax** [comment]
ci:forward

**Comment by MarinhoFeliphe** [comment]
ci:forward

**Comment by MarinhoFeliphe** [comment]
ci:stop

**Comment by MarinhoFeliphe** [comment]
ci:forward

**Pull request closed without merging**